File: Easy/1_two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<pair<int,int>>pairNums;
        int left = 0;

        for(int i = 0; i < nums.size(); i++){
            pairNums.push_back(make_pair(nums[i], i));
        }

        int right = pairNums.size()-1;
        int sum;
        sort(pairNums.begin(), pairNums.end());
        while( left < right){
            sum = pairNums[left].first + pairNums[right].first;
            if(sum == target) {
                return {pairNums[left].second, pairNums[right].second};
            }
            else if(sum > target) right--;
            else left++;
        }
        return {};
    }
};

File: python/easy/1_two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        l = len(nums)
        dic = {}
        for i in range(l):
            x = target - nums[i]
            if nums[i] in dic:
                return [dic[nums[i]], i]
            else:
                dic[x] = i
